mathesar_ui/src/systems/data-explorer/QueryManager.ts

```diff
@@ -196,17 +206,39 @@ export default class QueryManager extends QueryRunner {
     }
   }
 
+  private async reconcileDisplayOptions(serverResponse: ExplorationResult) {
+    const reconciliation = reconcileDisplayOptionsWithServerResponse(
+      this.getQueryModel().display_options,
+      serverResponse,
+    );
+    if (!reconciliation.hasChanged) return;
+
+    const newDisplayOptions = reconciliation.newValue;
+    const shouldAutoSave = !get(this.queryHasUnsavedChanges);
```

Comment:
The only scenario I could think of, where `shouldAutoSave` would be true here, is if the column type changes on the database. Is there any other scenario where this would be true?